src/highdicom/sr/utils.py

```diff
@@ -174,3 +175,169 @@ def get_coded_value(item: Dataset) -> CodedConcept:
         meaning=value.CodeMeaning,
         scheme_version=value.get('CodingSchemeVersion', None)
     )
+
+
+def get_coded_modality(sop_class_uid: str) -> Code:
```

Comment:
I have previously suggested to only pass the SOP Class UID rather than the full dataset (see https://github.com/MGHComputationalPathology/highdicom/pull/83#discussion_r652170213). That may be not such a good idea after all. There may be situation where we need more attributes than the SOP Class UID to determine the modality. For example, the modality of a Parametric Map depends on the modality of the source images.

Reply:
However, I think you intend this function to be used primarily to determine the modality of referenced images (i.e., using [Common Instance Reference Module Attributes](http://dicom.nema.org/medical/dicom/current/output/chtml/part03/sect_C.12.2.html#table_C.12-8) in which case this wouldn't work either. Maybe it would be better to keep this function private in `highdicom.sr.templates` for now..